src/client/refresh/soft/sw_image.c

```diff
@@ -708,8 +711,57 @@ R_InitImages (void)
 		// code never returns after ERR_FATAL
 		return;
 	}
-	memcpy(d_16to8table, table16to8, 0x10000);
-	ri.FS_FreeFile((void *)table16to8);
+
+	if (table16to8)
+	{
+		// Use predefined convert map
+		memcpy(d_16to8table, table16to8, 0x10000);
+		ri.FS_FreeFile((void *)table16to8);
+	}
+	else
+	{
+		// create new one
+		unsigned int r, g, b, i;
+
+		for (r = 0; r < 32; r++)
+		{
+			for (g = 0; g < 64; g++)
+			{
+				for (b = 0; b < 32; b++)
+				{
+					float diff = 1024;
+
+					for (i = 0; i < 256; i ++)
+					{
+						int r_c, g_c, b_c;
+						float curr_diff;
+
+						r_c = d_8to24table[i * 4 + 0] - (r << 3);
+						g_c = d_8to24table[i * 4 + 1] - (g << 2);
+						b_c = d_8to24table[i * 4 + 2] - (b << 3);
+
+						curr_diff = sqrt(
```

Comment:
There's no need to sqrt here since you only care about their relative values and not the actual result.

Reply:
Thank you, fixed